src/utils/formUtil.ts

```diff
@@ -278,8 +278,11 @@ export const eachComponentDataAsync = async (
           return true;
         }
         if (isComponentModelType(component, 'dataObject')) {
-          // No need to bother processing all the children data if there is no data for this form.
-          if (has(data, component.path)) {
+          // No need to bother processing all the children data if there is no data for this form or the reference value has not been loaded.
+          const nestedFormValue: any = get(data, component.path);
+          const noReferenceAttached = nestedFormValue && nestedFormValue._id && isEmpty(nestedFormValue.data) && !has(nestedFormValue, 'form');
+          const shouldProcessNestedFormData = nestedFormValue && nestedFormValue._id ? !noReferenceAttached : has(data, component.path);
+          if (shouldProcessNestedFormData) {
```

Comment:
small nit: use optional chaining
```suggestion
          const nestedFormValue: any = get(data, component.path);
          const noReferenceAttached = nestedFormValue?._id && isEmpty(nestedFormValue.data) && !has(nestedFormValue, 'form');
          const shouldProcessNestedFormData = nestedFormValue?._id ? !noReferenceAttached : has(data, component.path);
          if (shouldProcessNestedFormData) {
```

Reply:
changed